File: 043-MaxElementAndIndex/MaxElementAndIndex.py
def find_max_element(arr):
    max_element = arr[0]
    index = 0

    for i in range(1, len(arr)):
        if arr[i] > max_element:
            max_element = arr[i]
            index = i
    return max_element, index


arr = [1, 8, 4, 9, 6]
result = find_max_element(arr)
print("Maximum Element:", result[0])
print("Index:", result[1])

File: 043-MaxElementAndIndex/MaxElementAndIndex.cpp
#include<iostream>
#include<vector>
#include<utility> // for std::pair
using namespace std;

pair<int, int> MaxElementAndIndex(const vector<int>& arr) {
    if (arr.empty()) {
        return make_pair(-1, -1); // Return -1 for both if the array is empty
    }

    int maxVal = arr[0];
    int maxIndex = 0;

    for (int i = 1; i < arr.size(); i++) {
        if (arr[i] > maxVal) {
            maxVal = arr[i];
            maxIndex = i;
        }
    }
    return make_pair(maxVal, maxIndex);
}
int main() {
    vector<int> arr = {1, 8, 4, 9, 6};
    pair<int, int> result = MaxElementAndIndex(arr);
    
    cout << "Maximum Element: " << result.first << endl;
    cout << "Index: " << result.second << endl;
    
    return 0;
}